src/virt/mod.rs

```diff
@@ -19,7 +19,7 @@ pub enum ExecutionMode {
 }
 
 /// The context of a virtual firmware.
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
```

Comment:
Ok so that's a big NO-GO ^^

Copy is super expensive for big structs, and also super bug prone for anything that is not more than an integer wrapper.

What will happen if we add Copy is that each time we put a `VirtContext` in a new variable the compiler will copy it (instead of using the move semantic). So not only we might have bugs because we update the wrong copy now, but the struct is huge so it will have a huge performance impact in terms of memcopy. 

If the problem is the mutex, see the comment there to fix the issue.